### PR TITLE
feat: adds media translation samples

### PR DESCRIPTION
**Pull request by telpirion**
This pull request adds the following code snippet region tags:

- media_translation_translate_from_file
- media_translation_translate_from_mic

### CODE REVIEW

File: media-translation/cloud-client/translate_from_file.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+
+def translate_from_file(file_path):
+    from google.cloud import mediatranslation
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+
+    config = mediatranslation.StreamingTranslateSpeechConfig(
+        audio_config=mediatranslation.TranslateSpeechConfig(
+            audio_encoding='linear16',
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Why is this a string? 
It should be an enum like this: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/speech/cloud-client/transcribe_model_selection.py#L43

File: media-translation/cloud-client/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+google-cloud-media-translation>=0.1.0
```
Comment (nnegrey) [inline media-translation/cloud-client/requirements.txt]:
```suggestion
google-cloud-media-translation==0.1.0
```
use the explicit version

File: media-translation/cloud-client/translate_from_file.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Delete this

File: media-translation/cloud-client/translate_from_file.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+# Copyright 2020 Google Inc. All Rights Reserved.
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
# Copyright 2020 Google LLC
```

File: media-translation/cloud-client/translate_from_file.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+"""Google Cloud Media Translation sample application.
+Example usage:
+    python translate_from_file.py resources/audio.raw
+"""
+
+import argparse
+
+# [START media_translation_translate_from_file]
+
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
from google.cloud import mediatranslation
```

File: media-translation/cloud-client/translate_from_file.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+def translate_from_file(file_path):
+    from google.cloud import mediatranslation
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
```

File: media-translation/cloud-client/translate_from_file.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+Example usage:
+    python translate_from_file.py resources/audio.raw
+"""
+
+import argparse
+
+# [START media_translation_translate_from_file]
+
+
+def translate_from_file(file_path):
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
def translate_from_file(file_path='path/to/you/file.raw'):
```

File: media-translation/cloud-client/translate_from_file.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    from google.cloud import mediatranslation
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+
+    config = mediatranslation.StreamingTranslateSpeechConfig(
+        audio_config=mediatranslation.TranslateSpeechConfig(
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Let's pull this out and create a separate variable for this for readability

File: media-translation/cloud-client/translate_from_file.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    audio_content=chunk,
+                    streaming_config=config)
+
+    requests = request_generator(config, file_path)
+    responses = client.streaming_translate_speech(requests)
+
+    for response in responses:
+        # Once the transcription settles, the response contains the
+        # is_final result. The other results will be for subsequent portions of
+        # the audio.
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Do you need to set something in the config for partial transcription?

File: media-translation/cloud-client/translate_from_file.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        print(u'Partial recognition result: {0}'.format(source))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('stream', help='File to stream to the API')
+    args = parser.parse_args()
+    translate_from_file(args.stream)
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
```

File: media-translation/cloud-client/translate_from_file_test.py [hunk 11]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2020, Google, Inc.
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file_test.py]:
```suggestion
# Copyright 2020 Google LLC
```

File: media-translation/cloud-client/translate_from_file_test.py [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+RESOURCES = os.path.join(os.path.dirname(__file__), 'resources')
+
+
+def test_translate_streaming(capsys):
+    translate_from_file.translate_from_file(
+        os.path.join(RESOURCES, 'audio.raw'))
+    out, err = capsys.readouterr()
+
+    assert re.search(r'Partial translation', out, re.DOTALL | re.I)
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file_test.py]: This seems fine, you can also just do : `assert 'Partial translation' in out`

File: media-translation/cloud-client/translate_from_mic.py [hunk 13]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]:
```suggestion
```

File: media-translation/cloud-client/translate_from_mic.py [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+# Copyright 2020 Google Inc. All Rights Reserved.
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]:
```suggestion
# Copyright 2020 Google LLC
```

File: media-translation/cloud-client/translate_from_mic.py [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cloud Media Translation API sample application using a microphone.
+
+NOTE: This module requires the additional dependency `pyaudio`. To install
+using pip:
+
+    pip install pyaudio
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]: Added to `requirements.txt` can delete this

File: media-translation/cloud-client/translate_from_mic.py [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+def main():
+    source_language_code = 'en-US'
+    target_language_code = 'es-ES'
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+    config = mediatranslation.StreamingTranslateSpeechConfig(
+        audio_config=mediatranslation.TranslateSpeechConfig(
+            audio_encoding='linear16',
+            source_language_code=source_language_code,
+            target_language_code=target_language_code))
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]:
```suggestion
            source_language_code='en-US',
            target_language_code='es-ES'))
```

File: media-translation/cloud-client/translate_from_mic.py [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+            print(u'Final recognition result: {0}'.format(source))
+            break
+
+        print(u'\nPartial translation: {0}'.format(translation))
+        print(u'Partial recognition result: {0}'.format(source))
+
+
+def main():
+    source_language_code = 'en-US'
+    target_language_code = 'es-ES'
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]:
```suggestion
```

File: media-translation/cloud-client/translate_from_file.py (hunk 1 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Acknowledged. I will raise that issue with product engineering.

File: media-translation/cloud-client/translate_from_file.py (hunk 9 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: No.

File: media-translation/cloud-client/translate_from_file_test.py (hunk 12 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file_test.py]: Good to know!

File: media-translation/cloud-client/translate_from_mic.py (hunk 17 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_mic.py]: Done

File: media-translation/cloud-client/translate_from_mic.py (hunk 16 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_mic.py]: Done.

File: media-translation/cloud-client/translate_from_mic.py (hunk 15 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_mic.py]: Done.

File: media-translation/cloud-client/translate_from_mic.py (hunk 14 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_mic.py]: Done.

File: media-translation/cloud-client/translate_from_file_test.py (hunk 11 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file_test.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 10 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done

File: media-translation/cloud-client/translate_from_file.py (hunk 8 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 7 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 6 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 5 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 4 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/translate_from_file.py (hunk 3 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file.py]: Done.

File: media-translation/cloud-client/requirements.txt (hunk 2 above)
Comment (telpirion) [reply media-translation/cloud-client/requirements.txt]: Done.

File: media-translation/cloud-client/translate_from_file.py [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cloud Media Translation sample application.
+
+Example usage:
+    python translate_from_file.py resources/audio.raw
+"""
+
+import argparse
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]:
```suggestion
```

File: media-translation/cloud-client/translate_from_file.py [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+# [START media_translation_translate_from_file]
+from google.cloud import mediatranslation
+
+
+def translate_from_file(file_path='path/to/your/file'):
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+
+    speech_config = mediatranslation.TranslateSpeechConfig(
+        audio_encoding='linear16',
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Did this get resolved or is it still a string?

File: media-translation/cloud-client/translate_from_file_test.py [hunk 20]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2020, Google LLC
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file_test.py]:
```suggestion
# Copyright 2020 Google LLC
```

File: media-translation/cloud-client/translate_from_mic.py [hunk 21]
```diff
@@ -0,0 +1,156 @@
...
+"""
+
+# [START media_translation_translate_from_mic]
+from __future__ import division
+
+import itertools
+
+from google.cloud import mediatranslation
+import pyaudio
+from six.moves import queue
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_mic.py]: Does `six` need to be added to `requirments.txt`?
https://pypi.org/project/six/

File: media-translation/cloud-client/translate_from_file_test.py (hunk 20 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_file_test.py]: Done.

File: media-translation/cloud-client/translate_from_mic.py (hunk 21 above)
Comment (telpirion) [reply media-translation/cloud-client/translate_from_mic.py]: Done.

**Comment by nnegrey** [comment]
Blocked by: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/3116

File: media-translation/cloud-client/translate_from_file.py [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+
+# [START media_translation_translate_from_file]
+from google.cloud import mediatranslation
+
+
+def translate_from_file(file_path='path/to/your/file'):
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+
+    speech_config = mediatranslation.TranslateSpeechConfig(
```
Comment (nnegrey) [inline media-translation/cloud-client/translate_from_file.py]: Nit: so up to you, personally I like to have the variables names match the argument they'll be used in later, but the object name is `speech_config` 

```suggestion
    audio_config = mediatranslation.TranslateSpeechConfig(
```

Same thing for `config` below --> `streaming_config`, but again object name isn't the same.

**Pull request merged**